### DISCUSSION TITLE
Zip folder and publish release

### DISCUSSION THREAD

**Discussion by HonkingGoose** [semantic-release/semantic-release#2466] [Configuration] [answered]
Hi all! 👋 

I'm trying to get started with `semantic-release` for the first time. I've read the documentation but it's too complicated for me.

What I want to do:

1. Create a `.zip` of the `plain-language-guidelines` directory in my repository
1. Attach this `.zip` to a GitHub release
1. Create release notes based on the Conventional Commits in my repository
1. `semantic-release` should version and publish the release on GitHub, with GitHub Actions

Source repository:

https://github.com/HonkingGoose/vale-plain-language-guidelines

**Comment by travi** [comment, marked as answer]
1. i would normally recommend using a `prepack` script in node projects like yours when needing to build or process files before release, but since it sounds like you dont intend to publish to a registry, adding the [exec plugin](https://github.com/semantic-release/exec) to [your config](https://semantic-release.gitbook.io/semantic-release/usage/configuration#configuration-file) may be more appropriate
2. attaching your file to a github release would require [configuring the github plugin](https://github.com/semantic-release/github#options) to list your file in the `assets` list
3. this is default behavior of the github plugin, as long as you want those release notes defined in your github releases only. semantic-release does support updating a `CHANGELOG.md` in your repo, but it does add a fair amount of complexity, so i dont recommend it unless you really need it
4. you can find details about executing on github actions [here](https://semantic-release.gitbook.io/semantic-release/recipes/ci-configurations/github-actions#.github-workflows-release.yml-configuration-for-node-projects). also, since you wont need to publish to an npm registry, be sure to leave the npm plugin out of the plugins list in your config. it is enabled by default, which we intend to change in the future, but can be disabled by simply leaving in out of your config.

we know that our docs can always be improved, so if you had any specific stumbling points, we'd be happy to hear suggestions for improvements.

**Reply by HonkingGoose** [reply]
Thank you very much for giving me some pointers to get started. ❤️ 

1. I'm only publishing to GitHub releases. The Vale linter has a package registry for linter plugins, but I'm not going to bother with that (for now).
1. Thanks for the tip about configuring the GitHub plugin to list the file in the `assets` lists.
1. I'll stay away from using a `CHANGELOG.md` then, to keep it simple.
1. Yeah, I'm not publishing to npm, so I'll leave the npm plugin out of my list of plugins in my config.

> we know that our docs can always be improved, so if you had any specific stumbling points, we'd be happy to hear suggestions for improvements.

It's more of a general "where do I even start?" problem. The docs look nice, and seem very complete. There's just so much information, that it's hard to know where to start.

I'll go try to install and configure `semantic-release` with your pointers now. 🥳

**Reply by HonkingGoose** [reply]
I've followed the steps you listed, and I've created a PR to set up semantic release on my repository. I think I've got most of the things set up correctly, but I'm not sure enough to merge the PR.

Do you want to review my PR? It would help me very much to have you look it over. 😉 

- https://github.com/HonkingGoose/vale-plain-language-guidelines/pull/33

**Reply by HonkingGoose** [reply]
Hi @travi and @gr2m,

Do you want to review my PR, to setup `semantic-release` on my repository? It's my first time using `semantic-release` and I would like to get my config/setup reviewed by one of you guys. 😉 

I hope you're willing to help me out. Here's the link to the PR on my repository:

- https://github.com/HonkingGoose/vale-plain-language-guidelines/pull/33

In return, I'll run the Vale linter on your docs file in the repository and create a PR to fixup any bad stuff it finds. 😉 

Greetings,

HonkingGoose

---

**Edit:** Copy/pasted the correct link into the comment.

**Reply by travi** [reply]
i dont spot any obvious problems with your proposed config, so i would expect it to work. if you are concerned about getting the config working correctly before your first stable release, you could consider testing things out with a [pre-release workflow](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#prerelease) first

**Reply by HonkingGoose** [reply]
Thank you very much for looking over my configuration! ❤️ 

The pre-release workflow config looks complicated. I'll just start counting from `1.0.0` and create breaking releases when needed. 😄

**Reply by travi** [reply]
`beta` and `alpha` branches are configured to use the pre-release workflow by default, so as long as you configure semantic-release to execute on those branches, just creating the branch and following the usual commit convention would be enough
